src/backend/crossterm.rs

```diff
@@ -0,0 +1,287 @@
+                            CMouseButton::Left => MouseEvent::Press(MouseButton::Left),
+                            CMouseButton::Middle => {
+                                MouseEvent::Press(MouseButton::Middle)
+                            }
+                            CMouseButton::Right => {
+                                MouseEvent::Press(MouseButton::Right)
+                            }
+                            CMouseButton::WheelUp => MouseEvent::WheelUp,
+                            CMouseButton::WheelDown => MouseEvent::WheelDown,
+                        };
+
+                        if let MouseEvent::Press(btn) = event {
+                            self.last_button = Some(btn);
+                        }
+
+                        return Event::Mouse {
+                            event,
+                            position,
+                            offset: Vec2::zero(),
+                        }
+                    },
+                    CMouseEvent::Release(x, y)
+                    if self.last_button.is_some() =>
+                        {
+                            let event = MouseEvent::Release(self.last_button.unwrap());
+                            let position = (x - 1, y - 1).into();
+
+                            return Event::Mouse {
+                                event,
+                                position,
+                                offset: Vec2::zero(),
+                            }
+                        },
+                    CMouseEvent::Hold(x, y)
+                    if self.last_button.is_some() =>
+                        {
+                            let event = MouseEvent::Hold(self.last_button.unwrap());
+                            let position = (x - 1, y - 1).into();
+
+                            return Event::Mouse {
+                                event,
+                                position,
+                                offset: Vec2::zero(),
+                            }
+                        },
+                    _ => Event::Unknown(vec![]),
+                };
+            }
+            _ => Event::Unknown(vec![]),
+        };
+        Event::Unknown(vec![])
+    }
+}
+
+impl backend::Backend for Backend {
+    fn poll_event(&mut self) -> Option<Event> {
+        if let Some(input_event) = self.async_reader.next() {
+            Some(self.map_key(input_event))
+        }else {
+            None
+        }
+    }
+
+    fn finish(&mut self) {
+        self.cursor.goto(1,1);
+        self.terminal.clear(ClearType::All);
+        self.write(Attribute::Reset);
+        cursor().show();
+    }
+
+    fn refresh(&mut self) {
+        self.stdout.borrow_mut().flush().unwrap();
+    }
+
+    fn has_colors(&self) -> bool {
+        // TODO: color support detection?
+        true
+    }
+
+    fn screen_size(&self) -> Vec2 {
+        let size = self.terminal.terminal_size();
+        return Vec2::new(size.0 as usize + 1, size.1 as usize + 1);
```

Comment:
This could also be `Vec2::from(size) + (1, 1)`.

Reply:
Ok!